### PR TITLE
Bbaw egyptian

### PR DESCRIPTION
**Pull request by phiwi**
This is the "hieroglyph corpus" that I could unfortunately not contribute during the marathon. I re-extracted it again now, so that it is in the state as used in my paper (seee documentation). I hope it satiesfies your requirements and wish every scientist out their loads of fun deciphering a 5.000 years old language :-)

### CODE REVIEW

**Comment by albertvillanova** [comment]
Hi @phiwi,

Thanks for contributing this nice dataset. If you have any blocking problem or question, do not hesitate to ask here. We are pleased to help you.

Could you please first synchronize with our master branch? From you branch `bbaw_egyptian`, type:
```
git fetch upstream master
git merge upstream/master
```

File: datasets/bbaw_egyptian/bbaw_egyptian.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        my_urls = self._URLS
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN, gen_kwargs={"filepath": data_dir["all"]},
```
Comment (albertvillanova) [inline datasets/bbaw_egyptian/bbaw_egyptian.py:73]:
```suggestion
                name=datasets.Split.TRAIN,
                gen_kwargs={"filepath": data_dir["all"]},
```

Comment (albertvillanova) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: Could you please fix this minor quality code issue?

**Review by albertvillanova — COMMENTED** [review]
You should also remove the file `datasets/dummy/0.0.0/dummy_data.zip`, because you have already attached the dummy data in `datasets/bbaw_egyptian/dummy/0.0.0/dummy_data.zip`.

File: datasets/bbaw_egyptian/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+licenses:
+- cc-by-4.0
+multilinguality:
+- multilingual
+size_categories:
+- 100K<n<1000K
+source_datasets:
+- extended|wikipedia
+task_categories:
+- translation
```
Comment (lhoestq) [inline datasets/bbaw_egyptian/README.md:17]: `specialized egyptologists` is not a valid `annotations_creators` tag. You can use this instead:
```
annotations_creators:
- expert-generated
```
There is a tool to create those tags [here](https://huggingface.co/datasets/tagging/).

For the languages, there should be one language per line:
```
- de
- en
- eg
```

Finally the `task_ids` tags are missing:
```
task_categories:
- conditional-text-generation
task_ids:
- machine-translation
```

File: datasets/bbaw_egyptian/README.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  title = {Multi-Task Modeling of Phonographic Languages: Translating Middle Egyptian Hieroglyphs},
+  author = {Wiesenbach, Philipp and Riezler, Stefan},
+  journal = {Proceedings of the International Workshop on Spoken Language Translation},
+  journal-abbrev = {IWSLT},
+  year = {2019},
+  url = {https://www.cl.uni-heidelberg.de/statnlpgroup/publications/IWSLT2019_v2.pdf}
+}
+```
+
+### Contributions
```
Comment (lhoestq) [inline datasets/bbaw_egyptian/README.md:169]:
```suggestion
### Contributions

Thanks to [@phiwi](https://github.com/phiwi) for adding this dataset.
```

File: datasets/bbaw_egyptian/bbaw_egyptian.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        my_urls = self._URLS
+        data_dir = dl_manager.download_and_extract(my_urls)
```
Comment (lhoestq) [inline datasets/bbaw_egyptian/bbaw_egyptian.py:70]: There's no extraction
```suggestion
        data_dir = dl_manager.download(my_urls)
```

**Review by lhoestq — COMMENTED** [review]
Thanks for adding this one :) 

I left a few comments
Also could you remove the file at `datasets/dummy/0.0.0/dummy_data.zip` please ?

File: datasets/bbaw_egyptian/bbaw_egyptian.py (hunk 1 above)
Comment (phiwi) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: I changed the code as desired and commited it to my repo. Do I need to initiate a new pull request?

Comment (albertvillanova) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: This is still not fixed.

Comment (albertvillanova) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: No, just push:
```
git push origin bbaw_egyptian
```

Comment (phiwi) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: Okay, did so.

Comment (albertvillanova) [reply datasets/bbaw_egyptian/bbaw_egyptian.py:73]: This is the only remaining issue causing a test fail. Once fixed, all tests will pass.

Comment (phiwi) [reply datasets/bbaw_egyptian/bbaw_egyptian.py]: Ah, this happened by my autoformatting with black.

**Comment by lhoestq** [comment]
Thanks ! Can you check that you have `black==21.4b0` and run `make style` again ? This should fix the "check_code_quality" CI issue

**Comment by phiwi** [comment]
Reformatted with black.

File: datasets/bbaw_egyptian/bbaw_egyptian.py (hunk 1 above)
Comment (phiwi) [reply datasets/bbaw_egyptian/bbaw_egyptian.py]: @albertvillanova Has everything worked out correctly by now?

**Comment by albertvillanova** [comment]
Hi @phiwi, there are still some minor problems in relation with the tags you used in the dataset card (README.md).

Here you can find the output of the metadata validator:
```
WARNING:root:❌ Failed to validate 'datasets/bbaw_egyptian/README.md':
Could not validate the metada, found the following errors:
* field 'size_categories':
	['100K<n<1000K'] are not registered tags for 'size_categories', reference at https://github.com/huggingface/datasets/tree/master/src/datasets/utils/resources/size_categories.json
* field 'task_ids':
	['machine translation'] are not registered tags for 'task_ids', reference at https://github.com/huggingface/datasets/tree/master/src/datasets/utils/resources/tasks.json
* field 'languages':
	['eg'] are not registered tags for 'languages', reference at https://github.com/huggingface/datasets/tree/master/src/datasets/utils/resources/languages.json

```

**Comment by phiwi** [comment]
@albertvillanova corrected :-)

**Comment by albertvillanova** [comment]
Thanks, @phiwi. Now all tests should pass green.

However, I think there is still an issue with the language code:
- the code for the Ancient Egyptian is not `ar-EG`
- there is no ISO 639-1 code for the Ancient Egyptian
- there is an ISO 639-2 code: `egy`; but this code will not pass the validation test because it is not in the list of valid codes

I am not sure what to do in this case... Maybe @lhoestq has an idea?

**Comment by albertvillanova** [comment]
I have just checked that in the [list of valid codes](https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/languages.json) there are already ISO 639-2 codes. Therefore, I would suggest you to add it to the list:
```
"egy": "Egyptian (Ancient)",
```

**Comment by phiwi** [comment]
Done.

**Comment by phiwi** [comment]
Hope, everything is okay right now.

**Review by albertvillanova — APPROVED** [review]
It looks good to me. Let's see if @lhoestq has any other suggestions before merging it to master.

**Review by lhoestq — APPROVED** [review]
Looks all good now thanks !

**Pull request merged**